src/sas/qtgui/Plotting/Slicers/ArcInteractor.py

```diff
@@ -0,0 +1,135 @@
+    """
+    def __init__(self, base, axes, color='black', zorder=5, r=1.0,
+                 theta=np.pi / 3, phi=np.pi / 8):
+        BaseInteractor.__init__(self, base, axes, color=color)
+        self.markers = []
+        self.axes = axes
+        self.color = color
+        # Variables for the current mouse position
+        self._mouse_x = r
+        self._mouse_y = 0
+        # Last known mouse position, for when the cursor moves off the plot
+        self._save_x = r
+        self._save_y = 0
+        self.scale = 10.0
+        # Key variables for drawing the interactor element
+        self.theta = theta
+        self.phi = phi
+        self.radius = r
+        # Calculate the marker coordinates and define the marker
+        marker_x = self.radius * np.cos(theta - 0.5 * phi)
+        marker_y = self.radius * np.sin(theta - 0.5 * phi)
+        self.marker = self.axes.plot([marker_x], [marker_y], linestyle='',
+                                     marker='s', markersize=10,
+                                     color=self.color, alpha=0.6, pickradius=5,
+                                     label='pick', zorder=zorder,
+                                     visible=True)[0]
+        # Define the arc
+        [self.arc] = self.axes.plot([], [], linestyle='-', marker='', color=self.color)
+        # The number of points that make the arc line
+        self.npts = 40
+        # Flag to keep track of motion
+        self.has_move = False
+        self.connect_markers([self.marker, self.arc])
+        self.update()
+
+    def set_layer(self, n):
+        """
+        Allow adding plot to the same panel
+        :param n: the number of layer
+        """
+        self.layernum = n
+        self.update()
+
+    def clear(self):
+        """
+        Clear this slicer and its markers
+        """
+        self.clear_markers()
+        try:
+            self.marker.remove()
+            self.arc.remove()
+        except:
+            # Old version of matplotlib
+            for item in range(len(self.axes.lines)):
+                del self.axes.lines[0]
+
+    def update(self, theta=None, phi=None, r=None):
+        """
+        Draw the new roughness on the graph.
+        :param theta: angle from x-axis of the central point on the arc
+        :param phi: angle from the centre point on the arc to each of its edges
+        :param r: radius from (0,0) of the arc on a data2D plot
+        """
+        x = []
+        y = []
+        if theta is not None:
+            self.theta = theta
+        if phi is not None:
+            self.phi = phi
+        if r is not None:
+            self.radius = r
+        # Calculate the points on the arc, and draw them
+        for i in range(self.npts):
+            angleval = 2 * self.phi / (self.npts - 1) * i + (self.theta - self.phi)
+            xval = 1.0 * self.radius * np.cos(angleval)
+            yval = 1.0 * self.radius * np.sin(angleval)
+            x.append(xval)
+            y.append(yval)
+
```

Comment:
this can be improved (performance) by taking out the factor and offset calculations out of the loop.
precalculate
```
angle_factor = 2 * self.phi / (self.npts - 1)
angle_offset = self.theta - self.phi
```
and just use those in the loop, e.g. with
```
x = [self.radius * np.cos(angle_factor * i + angle_offset) for i in range(self.npts)]
y = [self.radius * np.sin(angle_factor * i + angle_offset) for i in range(self.npts)]
```